lib/Dialect/FIRRTL/Transforms/InferWidths.cpp

```diff
@@ -402,14 +418,306 @@ static void solveBinary(BinaryExpr *expr,
   if (expr->lhs()->solution.hasValue() && expr->rhs()->solution.hasValue())
     expr->solution =
         f(expr->lhs()->solution.getValue(), expr->rhs()->solution.getValue());
```

Comment:
What is f?

Reply:
It's the callback that actually goes ahead and performs the computation. I've renamed the argument to `callback` to make it easier to read.